modules/avformat/avformat.c

```diff
@@ -203,7 +204,20 @@ int avformat_shared_alloc(struct shared **shp, const char *dev)
 	if (err)
 		goto out;
 
-	ret = avformat_open_input(&st->ic, dev, NULL, NULL);
+	avdevice_register_all();
```

Comment:
can you please move this one to `module_init` ?

Reply:
Done, I was not sure where it needs to be.